### PR TITLE
Create ViewPropTypes

### PR DESCRIPTION
**Pull request by newtonry**
Due to the deprecation of ```View.propTypes```, we are now supposed to use ```ViewPropTypes``` imported from ```react-native```. This library doesn't support that yet, so this PR adds ```ViewPropTypes```. It should resolve issues like https://github.com/RealOrangeOne/react-native-mock/issues/139 and https://github.com/RealOrangeOne/react-native-mock/issues/132. In it, I've:
- Removed ```propTypes``` from being defined in ```View.js```
- Created ```ViewPropTypes``` which matches https://github.com/facebook/react-native/blob/master/Libraries/Components/View/ViewPropTypes.js minus ```TVViewPropTypes```
- I've also moved ```AccessibilityTraits``` and ```AccessibilityComponentTypes``` to a ```ViewAccessibility``` file. Let me know if this should go somewhere else.
- Expose ```ViewPropTypes```

### CODE REVIEW

File: src/components/View.js [hunk 1]
```diff
@@ -2,47 +2,17 @@
...
-  'button',
-  'radiobutton_checked',
-  'radiobutton_unchecked',
-];
-
-// TODO(lmr):
 const forceTouchAvailable = false;
 
 const statics = {
+  AccessibilityComponentType: AccessibilityComponentTypes,
```
Comment (parshap) [inline src/components/View.js:14]: Could use the property shorthand here.

**Review by parshap — APPROVED** [review]

File: src/components/View.js (hunk 1 above)
Comment (newtonry) [reply src/components/View.js:14]: There's a plurality difference here unfortunately. I made the decision to follow the pattern in ```react-native``` (https://github.com/facebook/react-native/blob/master/Libraries/Components/View/ViewAccessibility.js#L59-L64). However, I'm not super insightful about this property in ```statics```, so I'm open to ideas.

Comment (parshap) [reply src/components/View.js:14]: Ah -- missed that. Makes sense!

**Review by RealOrangeOne — APPROVED** [review]
Great work! Looks like there's some issues in the tests on node 4, but these seem to do with dependancy issues outside this PR, so i'm happy to merge as-is.

**Comment by jcharlet** [comment]
Hi guys, 
sorry for the newbie question but how to build my project with your master branch?

I tried in my project
```
npm install --save github:RealOrangeOne/react-native-mock#master
cd node_modules/react-native-mock/
npm install
```
but it must be wrong because my test still fails, I get 
>    Cannot find module 'setupDevtools' from 'setup.js'
>       
>       at Resolver.resolveModule (node_modules/jest-resolve/build/index.js:179:17)
>       at Object.<anonymous> (node_modules/react-native/jest/setup.js:30:1)
while in https://github.com/facebook/jest/issues/3822 they advise to remove the npm modules I just installed manually..

**Comment by jcharlet** [comment]
Hi guys,
sorry for the newbie question but how to load the master branch of react-native-mock in my project?

I tried
```
npm install --save github:RealOrangeOne/react-native-mock#master
cd nodes_modules/react-native-mock
npm install
rm -rf nodes_modules
```
but it must be wrong because I get this issue after:

>   ● Test suite failed to run
> 
>     TypeError: Cannot read property 'shape' of undefined
>       
>       at Object.<anonymous> (node_modules/react-native-mock/build/propTypes/EdgeInsetsPropType.js:8:33)
>       at Object.<anonymous> (node_modules/react-native-mock/build/propTypes/ViewPropTypes.js:6:25)
>       at Object.<anonymous> (node_modules/react-native-mock/build/components/View.js:7:20)

Thanks!

**Comment by newtonry** [comment]
@jcharlet if I understand correctly you shouldn't be ```rm -rf nodes_modules``` in the ```nodes_modules/react-native-mock``` directory. Those are the dependencies that ```react-native-mock``` has. By deleting them, ```React```(https://github.com/RealOrangeOne/react-native-mock/blob/master/src/propTypes/EdgeInsetsPropType.js#L4) is probably becoming ```undefined``` and hence, ```PropTypes.shape``` will not work.

Try it again without deleting the ```node_modules``` directory after installing the modules.

**Comment by jcharlet** [comment]
@newtonry great thanks! I have many warnings mentioning that I have duplicates of modules in my project and react-native-mock but it now works :) !

**Comment by jcharlet** [comment]
@newtonry great thanks! I had conflicts between react-native module of my project and react-native-mock (they use different versions). so I simply removed react-native from react-native-mock/node_modules (but still kept the rest) and it now works :) !

**Comment by gardner** [comment]
I am running into issues with this PR as well.

After running:
`npm install --save-dev github:RealOrangeOne/react-native-mock#master`

Then `npm run ava`

Looks like the `package.json` and `mock.js` expect a file to be present: `Error: Cannot find module './build/react-native'`

```
$ grep -H build/react-native node_modules/react-native-mock/*
node_modules/react-native-mock/mock.js:const ReactNativeMock = require('./build/react-native');
node_modules/react-native-mock/package.json:  "main": "build/react-native.js",
```

but there is no `build/` directory relative to the files:

```
$ls node_modules/react-native-mock
LICENSE      README.md    mock.js      package.json src
```

**Comment by lucascarvalho** [comment]
@gardner same problem here, did you find a solution for this?

**Comment by khirakawa** [comment]
@gardner @lucascarvalho You need to build `react-native-mock` if you install it via `npm install --save-dev github:RealOrangeOne/react-native-mock#master`

1. `cd node_modules/react-native-mock`
2. Add the following to `package.json`

  ```
  "babel": {
    "presets": [
      "airbnb",
      "react-native"
    ]
  },
  ```

3. Run `npm run build`

**Pull request merged**